examples/08_experimental/plot_new_surface_bids_analysis_experimental.py

```diff
@@ -0,0 +1,178 @@
+# in the :term:`BIDS` dataset.
+# To get the first level models we only have to specify the dataset directory
+# and the task_label as specified in the file names.
+from nilearn.glm.first_level import first_level_from_bids
+
+task_label = "languagelocalizer"
+models, run_imgs, events, confounds = first_level_from_bids(
+    data.data_dir,
+    task_label,
+    img_filters=[("desc", "preproc")],
+    hrf_model="glover + derivative",
+    n_jobs=2,
+)
+
+
+# %%
+# Project :term:`fMRI` data to the surface and compute GLM and contrasts
+#
+# The projection function simply takes the :term:`fMRI` data and the mesh.
+# Note that those correspond spatially, as they are both in :term:`MNI` space.
+from pathlib import Path
+
+from nilearn import surface
+from nilearn.experimental.surface import SurfaceImage, load_fsaverage
+
+fsaverage5 = load_fsaverage()
+
+# Empty lists in which we are going to store activation values.
+z_scores_left = []
+z_scores_right = []
+for first_level_glm, fmri_img, confound, event in zip(
+    models, run_imgs, confounds, events
+):
+    print(f"Running GLM on {Path(fmri_img[0]).relative_to(data.data_dir)}")
+
+    texture_left = surface.vol_to_surf(
+        fmri_img[0], fsaverage5["pial"].parts["left"]
+    )
+    texture_right = surface.vol_to_surf(
+        fmri_img[0], fsaverage5["pial"].parts["right"]
+    )
+    image = SurfaceImage(
+        mesh=fsaverage5["pial"],
+        data={
+            "left": texture_left.T,
+            "right": texture_right.T,
+        },
+    )
+
+    # Fit GLM.
+    first_level_glm.fit(run_imgs=image, events=event[0], confounds=confound[0])
+
+    # Contrast specification
+    design_matrix = first_level_glm.design_matrices_[0]
+    contrast_values = (design_matrix.columns == "language") * 1.0 - (
+        design_matrix.columns == "string"
+    )
+    z_scores = first_level_glm.compute_contrast(contrast_values, stat_type="t")
+    z_scores_left.append(z_scores.data.parts["left"])
+    z_scores_right.append(z_scores.data.parts["right"])
+
+
+# %%
+# Group study
+# -----------
+#
+# Individual activation maps have been accumulated
+# in the ``z_score_left`` and ``z_scores_right`` lists respectively.
+# We can now use them in a group study (one-sample study).
+#
+# Prepare figure for concurrent plot of individual maps
+# compute population-level maps for left and right hemisphere
+# We directly do that on the value arrays.
+import numpy as np
+from scipy.stats import norm, ttest_1samp
+
+_, pval_left = ttest_1samp(np.array(z_scores_left), 0)
+_, pval_right = ttest_1samp(np.array(z_scores_right), 0)
+
+# %%
+# What we have so far are p-values: we convert them to z-values for plotting.
+z_val_left = norm.isf(pval_left)
+z_val_right = norm.isf(pval_right)
+
+# %%
+# Plot the resulting maps, at first on the left hemisphere.
+from nilearn import plotting
+from nilearn.datasets import fetch_surf_fsaverage
+
+fsaverage = fetch_surf_fsaverage(mesh="fsaverage5")
+
+plotting.plot_surf_stat_map(
```

Comment:
- further refactoring will be possible once https://github.com/nilearn/nilearn/pull/4431 is merged